src/components/Icon/index.js

```diff
@@ -0,0 +1,63 @@
+import _ from 'underscore';
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import themeColors from '../../styles/themes/default';
+import variables from '../../styles/variables';
+import * as BrandAssets from './BrandAssets';
+import * as Expensicons from './Expensicons';
+
+const ICONS = _.extend(BrandAssets, Expensicons);
+
+const propTypes = {
+    /* The asset to render. */
+    icon: PropTypes.oneOf(_.values(ICONS)).isRequired,
+
+    /* The width of the icon. */
+    width: PropTypes.number,
+
+    /* The height of the icon. */
+    height: PropTypes.number,
+
+    /* Whether or not the icon is enabled. */
+    isEnabled: PropTypes.bool,
```

Comment:
I would expect the isEnabled to control visibility, but that doesn't seem to be the case? Is that what  `fillColor=undefined` does?

Reply:
Nope, `isEnabled` just changes the default color. For example, the pin icon on an unpinned chat is disabled, while the pin icon on a pinned chat is enabled.

`fillColor=undefined` makes it so that the fill color passed in doesn't affect full-color `svg` icons such as the Expensify.cash icon and make them monochrome.

Reply:
Okay, that makes more sense. but should the Icon Component understand the concept of disabled/enabled? IMO it makes more sense for the logic of enabled == darker to be done outside of the Icon class. The Icon class should be dumb and not know about any business logic IMO. 

Again, I think the argument against this is pretty reasonable though.

Reply:
Okay, yeah the motivation for having the concept of "enabled" icons was to prevent having to pass a fill color externally. But this ended up being sometimes unavoidable (we want some icons to have a white fill, and others to have an animated color, etc...) So now that we have a fill prop, we can nix the concept of "enabled" icons.

Expensicons have a default fill, while full-color brand assets do not - that hasn't changed (although I think the code is a bit cleaner now).